### ISSUE TITLE
Alternative reads found (under DP4) but no alternative base (ALT) reported

### ISSUE THREAD

**Issue by afclopes** [samtools/bcftools#1376]
Hello,

I am interested in getting the number of reads for the alternative and wild type base for mitochondrial DNA. However, I have found many circumstances where a significant number of alternative reads are reported, but no alternative base is reported. I was wondering if anyone could shed some light on why this could be happening, please?

Here is an example of the issue (code first, then one line of the result). As you can see, in the 5th column, ALT, no alternative bases are reported, however when looking at the values for DP4 (reference reads forward, reference reads reverse, alternative reads forward, alternative reads reverse) there seems to be significant numbers of alternative reads.

Code:
` bcftools mpileup -B -Ou --max-depth 1000 -Q 30 -f /path/"${file}"_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20.bam | bcftools call -m -Ov -o "${file}"_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_bcftools.vcf `

` bcftools norm -f /path/Genome_Downloads/GRCm38.p6.genome.fa "${file}"_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_bcftools.vcf > "${file}"_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_bcftools_Normalised.vcf `

Result:
> chrM    153     .       C       .       16.3113 .       DP=996;VDB=0.0368981;SGB=-0.693147;RPB=0.0866866;MQB=0.925709;MQSB=5.70557e-05;BQB=0.252241;MQ0F=0;AN=2;DP4=649,49,139,37;MQ=38 GT      0/0

Assuming these alternative reads are correct, what base are they referring to?
And why is there no alternative base reported?

While trying to figure this out, and after being prompted by Petr Daneck, I have tried to select just this line of interest and run the code with and without the parameter for -Q 30. This required me to do two new things that I was not doing before. First, to index the data before running the mpileup, second, to select just the line of interest. Below are my attempts:

Code with -Q 30:
` bcftools mpileup -B -Ou --max-depth 1000 -Q 30 -r chrM:153 -f /path/Genome_Downloads/GRCm38.p6.genome.fa SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20.bam | bcftools call -m -Ov -o SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_bcftoolsQ30.vcf `

` bcftools norm -f /raw/RawGenomicsE/afcl2/Genome_Downloads/GRCm38.p6.genome.fa SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_bcftools30.vcf > SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_bcftools30_Normalised.vcf `

Result:
> chrM    153     .       C       T       164     .       DP=1095;VDB=4.00463e-34;SGB=-0.693147;RPB=0.0875681;MQB=0.98382;MQSB=9.98142e-10;BQB=0.972589;MQ0F=0;ICB=1;HOB=0.5;AC=1;AN=2;DP4=546,97,164,60;MQ=38    GT:PL   0/1:198,0,255

Code without -Q 30:
` bcftools mpileup -B -Ou --max-depth 1000 -r chrM:153 -f /raw/RawGenomicsE/afcl2/Genome_Downloads/GRCm38.p6.genome.fa SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20.bam | bcftools call -m -Ov -o SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_bcftools.vcf `

` bcftools norm -f /raw/RawGenomicsE/afcl2/Genome_Downloads/GRCm38.p6.genome.fa SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_bcftools.vcf > SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_bcftools_Normalised.vcf `

Result:
> chrM    153     .       C       T       95      .       DP=1095;VDB=6.71328e-35;SGB=-0.693147;RPB=0.0871308;MQB=0.944941;MQSB=3.11837e-10;BQB=0.897481;MQ0F=0;ICB=1;HOB=0.5;AC=1;AN=2;DP4=556,100,170,61;MQ=38  GT:PL   0/1:129,0,255

As you can see, these two new lines of result do not match what I had previously found. Since I had never used this parameter of ‘-r’. I tried using the ‘-r’ again but this time to select all of the mitochondrial DNA (previously I had been selecting the mitochondria DNA using 'grep'), and when I look at that same position, again it is different from the latest two cases shown above (in terms of the reads found under DP4). Here it is:

Code:
` bcftools mpileup -B -Ou --max-depth 1000 -Q 30 -r chrM -f /raw/RawGenomicsE/afcl2/Genome_Downloads/GRCm38.p6.genome.fa SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20.bam | bcftools call -m -Ov -o SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_Mito_bcftoolsQ30.vcf `

` bcftools norm -f /raw/RawGenomicsE/afcl2/Genome_Downloads/GRCm38.p6.genome.fa SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_Mito_bcftools30.vcf > SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_Mito_bcftools30_Normalised.vcf `

Result:
> chrM    153     .       C       T       75      .       DP=996;VDB=0.0368981;SGB=-0.693147;RPB=0.0866866;MQB=0.925709;MQSB=5.70557e-05;BQB=0.252241;MQ0F=0;ICB=1;HOB=0.5;AC=1;AN=2;DP4=649,49,139,37;MQ=38      GT:PL   0/1:110,0,255

This time however, we can see exactly the same values for the reads under DP4 as we observe in the very first instance, yet, the ALT base shows as a ’T' instead of ‘.' .

I wonder if it is more reliable to use indexing before bcftools mpileup? Perhaps after indexing it is easier for bcftools to count all the reads?

And whether it is more reliable to use '-r chrM' instead of 'grep “chrM”' to select only the mitochondrial DNA? In order to address this last question, I tried one last test: using the indexed data, -Q 30, and using 'grep "chrM"' instead of '-r chrM'. Below is what I get:

Code:
` bcftools mpileup -B -Ou --max-depth 1000 -Q 30 -f /raw/RawGenomicsE/afcl2/Genome_Downloads/GRCm38.p6.genome.fa SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20.bam | bcftools call -m -Ov -o SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_NoMito_bcftoolsQ30.vcf `

` bcftools norm -f /raw/RawGenomicsE/afcl2/Genome_Downloads/GRCm38.p6.genome.fa SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_NoMito_bcftoolsQ30.vcf > SRR1411188_val.R1_bismark_bt2_pe.deduplicated_sorted_MAPQ20_NoMito_bcftoolsQ30_Normalised.vcf `

Result:
> chrM    153     .       C       .       16.3113 .       DP=996;VDB=0.0368981;SGB=-0.693147;RPB=0.0866866;MQB=0.925709;MQSB=5.70557e-05;BQB=0.252241;MQ0F=0;AN=2;DP4=649,49,139,37;MQ=38 GT      0/0

As you can see, we are back to finding in the place of an ATL base, a '.' instead of a 'T'. Which is exactly the same result as when I used no indexing (in the very first test). Suggesting then that the change of base in ALT could be due to the technique I use to select the mitochondrial DNA. But I cannot figure out why that would be... 
Also, we can see that with and without indexing, the same number of reads appears in the last two results and the very first result, suggesting that the varying number of reads does not seem to be to do with the indexing, but perhaps '-r'?

This brings me back to the recurrent questions:

Is '-r chrM' more reliable to use than 'grep "chrM"' to select only the mitochondrial DNA? Is 'grep' somehow causing the loss of information for column ALT? And can we assume that the alternative reads with no ALT base reported belong to T as we find using ‘-r chrM’?

In conclusion, I should then be using ‘-r chrM’, and inevitably also indexing? Or since using ‘-r chrM:153’ showed different numbers for DP4, it suggests that ‘-r’ is also not very reliable? In that case, I am unsure what to do now. Does anyone have any suggestions?

Apologies for the long issue post, but I thought the issue was a bit too complex to explain in few words.

Thank you for your help,
Amanda

**Comment by pd3** [comment]
Hi,
apologies for the belated answer. I believe this should solve your problem: try `bcftools mpileup -a AD`, this will add per-allele counts, and `bcftools call -mvA`, this will preserve all alternate alleles on output.

**Comment by alnam3** [comment]
Hi @pd3 I got the same problem, and used the options you recommended (along with -B which helped me having more snps recovered), but I still have missed snps. 
I compared the vcf files with and without the bcftools call -v option, and there are lines such as this one

`reg1        78      .       T       A,G,C   21.7174 .       DP=17908;VDB=0;SGB=-0.693147;RPB=0.0215026;MQB=0;BQB=0.835437;MQ0F=0;AC=0,0,0;AN=2;DP4=5257,0,3710,0;MQ=48      GT:PL   0/0:9,0,50,255,255,109,255,255,112,109`

when I output all positions (without the -v option) which do not appear in the -v file.
Any idea of why this happens?

I saw it was possible to change the threshold for an alternative variant to be called as such, but only with the -c option.

Thanks a lot!
